frontend/src/components/pongGame/pongGame.vue

```diff
@@ -0,0 +1,327 @@
+				heightStart,
+				lineWidth,
+				lineHeight,
+			);
+		},
+
+		//  https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/arc
+		drawBall(ball: Ball) {
+			const startingX: number = this.widthPercentage(ball.position.x);
+			const startingY: number = this.heightPercentage(ball.position.y);
+			const radius: number = this.widthPercentage(ball.radius);
+			const startAngle: number = 0;
+			const endAngle: number = Math.PI * 2; // full circle
+
+			this.context.beginPath();
+			this.context.arc(
+				startingX,
+				startingY,
+				radius,
+				startAngle,
+				endAngle,
+			);
+			this.context.lineWidth = this.widthPercentage(0.3);
+			this.context.stroke();
+		},
+
+		// https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/fillRect
+		drawPaddle(paddle: Paddle) {
+			const lineWidth: number = this.widthPercentage(1);
+			const lineHeight: number = this.heightPercentage(12);
+			const y: number =
+				this.heightPercentage(paddle.position.y) - lineHeight / 2; // middle of the canvas
+
+			// Player 1
+			let x = 0;
+			if (paddle.position.x === 100) {
+				// Player 2
+				x = this.width - lineWidth;
+			}
+			this.context.fillRect(x, y, lineWidth, lineHeight);
+		},
+
+		clearCanvas() {
+			this.context.clearRect(0, 0, this.width, this.height);
+		},
+
+		render(elementPositions: ElementPositions) {
+			this.clearCanvas();
+			this.drawMiddleLine();
+			this.drawPaddle(elementPositions.playerOnePaddle);
+			this.drawPaddle(elementPositions.playerTwoPaddle);
+			this.drawBall(elementPositions.ball);
+		},
+
+		keyDown(keyPress: KeyboardEvent) {
+			if (this.isPressed[keyPress.key] !== undefined) {
+				this.isPressed[keyPress.key] = true;
+			}
+		},
+
+		keyUp(keyPress: KeyboardEvent) {
+			if (this.isPressed[keyPress.key] !== undefined) {
+				this.isPressed[keyPress.key] = false;
+			}
+		},
+
+		resetPressedKeys() {
+			this.isPressed.ArrowDown = false;
+			this.isPressed.ArrowUp = false;
+			this.isPressed.w = false;
+			this.isPressed.s = false;
+		},
+		// MAIN GAME LOOP
+		getUpdatedPositions(timeStamp: DOMHighResTimeStamp) {
+			if (this.previousTimeStamp === 0) {
+				this.previousTimeStamp = timeStamp;
+			}
+			const elapsedTime = timeStamp - this.previousTimeStamp;
+			const intervalMs = 10; // refresh rate of a browser is 1/60th of a sec (17)
+			// redraws after intervalMs
+			if (elapsedTime > intervalMs) {
+				this.previousTimeStamp = timeStamp;
+				this.socket!.emit('updatePositions', this.isPressed);
+			}
+
+			if (this.gameStatus === GameStatusEnum.PLAYING) {
+				window.requestAnimationFrame(this.getUpdatedPositions);
+			} else if (this.gameStatus === GameStatusEnum.POINT_OVER) {
+				// update scores
+				this.gameStatus = GameStatusEnum.PLAYING;
+				this.resetPressedKeys();
+				// reset positions
+				this.socket.emit('resetAfterPointFinished');
+
+				// restart game loop
+				window.requestAnimationFrame(this.getUpdatedPositions);
+			}
+			//  else game is over
```

Comment:
Should here then the `finishGame` method be called? (I currently don't see it called anywhere, but it's late.. so that could be just me..)